File: proyectos/caso4/src/pedidos.py
"""Capacidad "pedidos". Reescrita desde Node.js guiada por
docs/specs/spec-pedidos.md (no por src/pedidos.js del legacy-node).
"""

import time

from . import store

ESTADOS_VALIDOS = ["pendiente", "enviado", "entregado", "cancelado"]


def crear_pedido(cliente=None, items=None):
    if not cliente:
        raise ValueError("cliente es obligatorio")
    if not items:
        raise ValueError("items no puede estar vacio")
    return store.agregar(
        {
            "cliente": cliente,
            "items": items,
            "estado": "pendiente",
            "creadoEn": int(time.time() * 1000),
        }
    )


def cambiar_estado(pedido_id, nuevo_estado):
    if nuevo_estado not in ESTADOS_VALIDOS:
        raise ValueError(f"estado invalido: {nuevo_estado}")
    pedido = store.buscar_por_id(pedido_id)
    if pedido is None:
        raise ValueError(f"pedido {pedido_id} no existe")
    pedido["estado"] = nuevo_estado
    return pedido

File: proyectos/caso5/src/pedidos.js
'use strict';

const store = require('./store');

const ESTADOS_VALIDOS = ['pendiente', 'enviado', 'entregado', 'cancelado'];

function crearPedido({ cliente, items } = {}) {
  if (!cliente) throw new Error('cliente es obligatorio');
  if (!Array.isArray(items) || items.length === 0) {
    throw new Error('items no puede estar vacío');
  }
  return store.agregar({
    cliente,
    items,
    estado: 'pendiente',
    creadoEn: Date.now(),
  });
}

function cambiarEstado(id, nuevoEstado) {
  if (!ESTADOS_VALIDOS.includes(nuevoEstado)) {
    throw new Error(`estado inválido: ${nuevoEstado}`);
  }
  const pedido = store.buscarPorId(id);
  if (!pedido) throw new Error(`pedido ${id} no existe`);
  pedido.estado = nuevoEstado;
  return pedido;
}

module.exports = { crearPedido, cambiarEstado, ESTADOS_VALIDOS };
